actionpack/lib/action_dispatch/middleware/static.rb

```diff
@@ -94,9 +98,9 @@ def gzip_file_path(path)
   # produce a directory traversal using this middleware. Only 'GET' and 'HEAD'
   # requests will result in a file being returned.
   class Static
-    def initialize(app, path, cache_control=nil)
+    def initialize(app, path, cache_control = nil, headers = {})
       @app = app
-      @file_handler = FileHandler.new(path, cache_control)
+      @file_handler = FileHandler.new(path, cache_control, headers)
```

Comment:
Since we're passing response headers to FileHandler now, we can deprecate the `cache_control` argument and handle it here rather than in `FileHandler#initialize`:
```ruby
# Static
def initialize(app, path, deprecated_cache_control = :not_set, headers: nil)
  if deprecated_cache_control != :not_set
    ActiveSupport::Deprecation.warn "The `cache_control = '…'` argument is deprecated, replaced by `headers: { 'Cache-Control' => '…' }`, and will be removed in Rails 5.1."
    headers = (headers || {}).merge('Cache-Control'.freeze => deprecated_cache_control)
  end

  @app = app
  @file_handler = FileHandler.new(path, headers: headers)
end

# FileHandler
def initialize(root, headers:)
  …
  @file_server = ::Rack::File.new(@root, headers)
end
```